Microsoft/accessibility-insights-windows
# Support GitHub Issue Filing for AI-Win

#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [x] Attach any screenshots / GIF's that are applicable.

Note - After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge. 

#### Describe the change
- Refactor AI-Win code to support multiple bug filing extensions.
- Support GitHub Issue Filing for AI-Win.
- Refactor Azure DevOps bug filing.

**Test cases (you can try more):**
1. User has not configured any reporter - startup.
2. User has configured any reporter - startup. Make sure the logo is updated when the user is signed in.
3. No issue reporters - play around.
4. Use the connection logo to get to the connections tab and play around. Configure a connection, save and exit. Use the settings button to go to the connections tab and make sure that you see the previously saved connection and the corresponding UI from the extension.
5. Reverse of 4. Set from Settings check from the logo.
6. Save with only application settings changes.
7. Save with only connection settings changes.
8. Save with both types of changes.
9. Switch  between the bug reporting options without saving and make sure that there is no impact.
10. Make sure that logo is updated when the issue reporting option changes (when saved).
11. File bug using all possible ways.     From Live inspect Hierarchy tree     From Automated checks Hierarchy tree (For issue and without issue)     From the how to fix ((For issue and without issue).

**Screenshots**
- No extension is configured.
![image](https://user-images.githubusercontent.com/45858632/55116809-cebf9a00-50a5-11e9-832f-1a00e106d2eb.png)
![image](https://user-images.githubusercontent.com/45858632/55116845-f3b40d00-50a5-11e9-9cfa-325321b3e1c6.png)
![image](https://user-images.githubusercontent.com/45858632/55116864-03cbec80-50a6-11e9-921a-2059931fc7d3.png)
![image](https://user-images.githubusercontent.com/45858632/55116874-0cbcbe00-50a6-11e9-9a06-5a86bcc59ef4.png)

- Extension is configured but none is selected
![image](https://user-images.githubusercontent.com/45858632/55116936-4a214b80-50a6-11e9-8196-f49599b8c94b.png)
![image](https://user-images.githubusercontent.com/45858632/55116988-72a94580-50a6-11e9-8882-686638dd29a7.png)
![image](https://user-images.githubusercontent.com/45858632/55117003-7ccb4400-50a6-11e9-8b15-e586b3b23abe.png)
![image](https://user-images.githubusercontent.com/45858632/55117010-85bc1580-50a6-11e9-800a-5d8e7fb012de.png)
![image](https://user-images.githubusercontent.com/45858632/55117020-8f457d80-50a6-11e9-8bfe-182c9c9d9315.png)
![image](https://user-images.githubusercontent.com/45858632/55117052-ad12e280-50a6-11e9-87dc-e0d9e65c11d7.png)
![image](https://user-images.githubusercontent.com/45858632/55117077-c1ef7600-50a6-11e9-9eb3-d67da56cf78b.png)
![image](https://user-images.githubusercontent.com/45858632/55117090-cb78de00-50a6-11e9-812a-c29d9701e17f.png)

- Multiple extensions are configured
![image](https://user-images.githubusercontent.com/45858632/55117171-109d1000-50a7-11e9-88ad-d8559bff1ce1.png)
![image](https://user-images.githubusercontent.com/45858632/55117177-1a267800-50a7-11e9-9a32-32319e9fe8bf.png)
![image](https://user-images.githubusercontent.com/45858632/55117217-3aeecd80-50a7-11e9-827e-55ce3290cf71.png)
![image](https://user-images.githubusercontent.com/45858632/55117229-46da8f80-50a7-11e9-8634-fccdc86386d0.png)

- Restore configuration
![image](https://user-images.githubusercontent.com/45858632/55117262-6c679900-50a7-11e9-8d3d-7aa7a83bd8aa.png)
![image](https://user-images.githubusercontent.com/45858632/55117291-87d2a400-50a7-11e9-998d-5d8d11a88221.png)

- Single Issue
![image](https://user-images.githubusercontent.com/45858632/55117340-bcdef680-50a7-11e9-9358-0a0871312210.png)

- No issue
![image](https://user-images.githubusercontent.com/45858632/55117370-da13c500-50a7-11e9-8a9a-eeda86420827.png)
